adapters/adgeneration/adgeneration.go

```diff
@@ -0,0 +1,264 @@
+package adgeneration
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type AdgenerationAdapter struct {
+	endpoint        string
+	version         string
+	defaultCurrency string
+}
+
+// Server Responses
+type adgServerResponse struct {
+	Locationid string        `json:"locationid"`
+	Dealid     string        `json:"dealid"`
+	Ad         string        `json:"ad"`
+	Beacon     string        `json:"beacon"`
+	Beaconurl  string        `json:"beaconurl"`
+	Cpm        float64       `jsons:"cpm"`
+	Creativeid string        `json:"creativeid"`
+	H          uint64        `json:"h"`
+	W          uint64        `json:"w"`
+	Ttl        uint64        `json:"ttl"`
+	Vastxml    string        `json:"vastxml,omitempty"`
+	LandingUrl string        `json:"landing_url"`
+	Scheduleid string        `json:"scheduleid"`
+	Results    []interface{} `json:"results"`
+}
+
+func (adg *AdgenerationAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	numRequests := len(request.Imp)
+	var errs []error
+
+	if numRequests == 0 {
+		errs = append(errs, &errortypes.BadInput{
+			Message: "No impression in the bid request",
+		})
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	requestArray := make([]*adapters.RequestData, 0, numRequests)
+
+	for index := 0; index < numRequests; index++ {
+		requestUri, err := adg.getRequestUri(request, index)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+		request := &adapters.RequestData{
```

Comment:
Local variable `request` has the same name as parameter `request *openrtb.BidRequest`. Can we rename one of them to tell them apart easier?